packages/office-ui-fabric-react/src/components/DetailsList/DetailsColumn.styles.ts

```diff
@@ -179,26 +193,12 @@ export const getStyles = (props: IDetailsColumnStyleProps): IDetailsColumnStyles
 
     accessibleLabel: [hiddenContentStyle],
 
-    borderAfterDropping: [
-      {
-        borderStyle: 'solid',
-        borderWidth: 1,
-        borderColor: palette.themePrimary,
-        left: -1,
-        lineHeight: 31
-      }
-    ],
+    borderWhileDragging: borderWhileDragging,
 
-    borderWhileDragging: [
-      {
-        selectors: {
-          '.ms-DetailsHeader &.ms-DetailsHeader-cell': {
-            borderStyle: 'solid',
-            borderWidth: 1,
-            borderColor: palette.themePrimary
-          }
-        }
-      }
-    ]
+    noBorderWhileDragging: [borderAfterDragOrDrop, { transition: 'border-color 0.2s ease' }],
+
+    borderAfterDropping: [borderWhileDragging, { left: -1, lineHeight: 31 }],
+
+    noBorderAfterDropping: [borderAfterDragOrDrop, { transition: 'border-color 1.5s ease' }]
```

Comment:
@Vitalius1 I feel like these selectors can be simplified... but I haven't quite figured it out yet. Any ideas?

Reply:
Btw, I'm curious.. are these fading borders in the `DetailsList` spec? How about for Fluent?

Reply:
I think @betrue-final-final would be able to answer this better.

Reply:
Perhaps, but @Vitalius1 _just_ made updates to `DetailsList` for Fluent (#7188) so he's probably aware of the spec also.

Reply:
As for @KevinTCoughlin question if there is a better way to do it to trigger the transition, I might have an idea. Wasn't the point of CSS-in-JS styling help reduce the number of classNames we register and leverage the conditional styling based on the state of the component? One example would be how I did this in `Shimmer` https://github.com/OfficeDev/office-ui-fabric-react/blob/72942a6152855a0717f6bcdb0f12309c5be5d25b/packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.base.tsx#L36-L52
This way your component's state controls what styles it needs. Also, I passed the interval as a styleProp to have it defined in one place to prevent changes in one place and not the other. In styles you would have something like this: https://github.com/OfficeDev/office-ui-fabric-react/blob/72942a6152855a0717f6bcdb0f12309c5be5d25b/packages/office-ui-fabric-react/src/components/Shimmer/Shimmer.styles.ts#L65-L86

Reply:
> quick question to @arkogupta would be, where are these numbers are coming from? From what it seems you should have a designer's signoff on these values. To be more exact, we have animation variables defined [here](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/styling/src/styles/AnimationStyles.ts), so I might think we should use them to be consistent, even though sometimes we have exceptions from that when the design dictates something we don't have available.

@Vitalius1 , these animation durations are coming from the spec for drag-n-drop for SharePoint.
Hi @betrue-final-final, you might want to add something here?

Reply:
@Vitalius1, thanks for sharing your example. Along similar lines, we can use a prop on dragStart to conditionally have the border, and use transition. But unfortunately that'd mean whenever that prop would change, i.e on dragEnd, the component would re-render, and that's a performance bottleneck we'd like to avoid. Same goes for the column being dropped case. 

Tagging @laxmikankanala, as she's the one who implemented the feature and might be able to provide some additional thoughts on the above.

As for passing the transition duration as a styleProp, that's a good idea and I've made the changes accordingly.

Reply:
I think passing in the transition durations is a reasonable compromise for now. Long-term it would be nice if we could remove the need for modifying the element's `classList` via timeouts. But since we already have that pattern in this component, we shouldn't block on it.

Reply:
_At least_ the durations will remain in-sync.